src/rsz/src/EstimateWireParasitics.cc

```diff
@@ -736,4 +742,184 @@ void Resizer::parasiticsInvalid(const dbNet* net)
   parasiticsInvalid(db_network_->dbToSta(net));
 }
 
+void Resizer::openSpefFile(const char* path)
+{
+  if (path == nullptr) {
+    write_spef_file = false;
+    return;
+  }
+
+  std::string file_path(path);
+  write_spef_file = !file_path.empty();
+  if (write_spef_file) {
+    for (sta::Corner* corner : *sta_->corners()) {
+      file_path = path;
+      std::string suffix("_");
+      suffix.append(corner->name());
+      if (file_path.find(".spef") != std::string::npos
+          || file_path.find(".SPEF") != std::string::npos) {
+        file_path.insert(file_path.size() - 5, suffix);
+      } else {
+        file_path.append(suffix);
+      }
+
+      spef_file[corner].open(file_path, std::ofstream::out);
+      writeSpefHeader(corner, db_network_);
+      writeSpefPorts(corner, db_network_);
+    }
+  }
+}
+
+void Resizer::closeSpefFile()
+{
+  if (write_spef_file) {
+    for (sta::Corner* corner : *sta_->corners()) {
+      spef_file[corner].close();
+    }
+    spef_file.clear();
+  }
+}
+
+void Resizer::writeSpefHeader(Corner* corner, dbNetwork* network)
+{
+  spef_file[corner] << "*SPEF \"ieee 1481-1999\"" << '\n';
+  spef_file[corner] << "*DESIGN \"" << network->block()->getName() << "\""
+                    << '\n';
+  spef_file[corner] << "*DATE \"11:11:11 Fri 11 11, 1111\"" << '\n';
+  spef_file[corner] << "*VENDOR \"The OpenROAD Project\"" << '\n';
+  spef_file[corner] << "*PROGRAM \"OpenROAD\"" << '\n';
+  spef_file[corner] << "*VERSION \"1.0\"" << '\n';
+  spef_file[corner] << "*DESIGN_FLOW \"NAME_SCOPE LOCAL\" \"PIN_CAP NONE\""
+                    << '\n';
+  spef_file[corner] << "*DIVIDER /" << '\n';
+  spef_file[corner] << "*DELIMITER :" << '\n';
+  spef_file[corner] << "*BUS_DELIMITER []" << '\n';
+  spef_file[corner] << "*T_UNIT 1 NS" << '\n';
+  spef_file[corner] << "*C_UNIT 1 PF" << '\n';
+  spef_file[corner] << "*R_UNIT 1 OHM" << '\n';
+  spef_file[corner] << "*L_UNIT 1 HENRY" << '\n';
+  spef_file[corner] << '\n';
+}
+
+void Resizer::writeSpefPorts(Corner* corner, dbNetwork* network)
+{
+  auto pin_iter = network->pinIterator(network->topInstance());
```

Comment:
sta is old fashioned and this pointer must be freed by the caller.  The easiest is to store the result value in a unique_ptr.